### PR TITLE
Support running SRO on vanilla k8s

### PR DESCRIPTION
**Pull request by dagrayvid**
This PR changes SRO to support running on vanilla k8s. For the openshift specific resources, we can use a new function `clients.HasResource` to check if it is available, and warn (but not fail) if it is not found, for instance,
```
clusterOperatorAvailable, err := clients.HasResource(configv1.SchemeGroupVersion.WithResource("clusteroperators"))
...
if clusterOperatorAvailable {
	// ... update status
} else {
	log.Info("Warning: ClusterOperator resource not available. Can be ignored on vanilla k8s.")
}

```

For the cases where the needed resource is hard to specify for example in SetUpWithManager in  controllers/specialresource_controller.go , we default to checking for the availability of the BuildConfigs API to determine whether or not we should use OCP specific resources.

This PR also adds a Make target to allow a user to configure the Deployment / kustomize .yaml for deploying on Vanilla, which just involves disabling the clusterOperator, prometheus ServiceMonitor and the kube-rbac-proxy used for the /metrics endpoint. In the case where a user wants to use these APIs and has them installed on a vanilla k8s cluster they can be enabled manually by commenting/uncommenting  lines in `config/default/kustomization.yaml`.

### CODE REVIEW

File: controllers/resources.go [hunk 1]
```diff
@@ -50,6 +51,14 @@ func getChartTemplates(r *SpecialResourceReconciler) (*unstructured.Unstructured
 
 func createImagePullerRoleBinding(r *SpecialResourceReconciler) error {
 
+	_, err := clients.Interface.RbacV1().ClusterRoles().Get(context.TODO(), "system:image-puller", metav1.GetOptions{})
```
Comment (zvonkok) [inline controllers/resources.go]: clients.Interface.Get did not work?

Comment (zvonkok) [reply controllers/resources.go]: Why is the handling not done in line 72?

File: controllers/runtime.go [hunk 2]
```diff
@@ -144,6 +144,15 @@ func retryGetPushSecretName(r *SpecialResourceReconciler) (string, error) {
 
 func getPushSecretName(r *SpecialResourceReconciler) (string, error) {
```
Comment (zvonkok) [inline controllers/runtime.go:153]: in vanilla k8s there are no push secrets?

File: controllers/resources.go (hunk 1 above)
Comment (dagrayvid) [reply controllers/resources.go]: Good point

File: controllers/runtime.go (hunk 2 above)
Comment (dagrayvid) [reply controllers/runtime.go:153]: I don't think there are. At least not in my kOps cluster

**Comment by dagrayvid** [comment]
@zvonkok I rebased, and fixed the issue with the missing volume. Tested on vanilla and the e2e tests pass, so from my side this PR is ready to merge.

**Pull request merged**